components/Wallet.tsx

```diff
@@ -1,63 +1,124 @@
 import WalletModal from "@/components/shared/WalletModal";
+import ThreeDots from "@/components/shared/ThreeDots";
+import AccountIdenticon from "@/components/shared/AccountIdenticon";
+
+type WalletState = "NotConnected" | "Connecting" | "Connected";
 
 const Wallet: React.FC<{}> = () => {
 	const [modalOpen, setModalOpen] = useState<boolean>(false);
-	const { selectedAccount } = useWallet();
+	const { selectedAccount, balances, connectWallet } = useWallet();
+	const walletState = useMemo<WalletState>(() => {
+		if (!selectedAccount) return "NotConnected";
+		if (selectedAccount && !balances?.length) return "Connecting";
+		return "Connected";
+	}, [selectedAccount, balances]);
+
+	const onWalletClick = useCallback(async () => {
+		if (walletState === "Connecting") return;
+		if (walletState === "Connected") return setModalOpen(true);
+
+		await connectWallet();
+		setModalOpen(true);
+	}, [walletState, connectWallet]);
 
 	return (
 		<>
-			{modalOpen && <WalletModal setModalOpen={setModalOpen} />}
-			<Frame
-				sx={{
-					"top": "3em",
-					"right": "3em",
-					"backgroundColor": modalOpen ? "#1130FF" : "#FFFFFF",
-					"cursor": "pointer",
-					"textAlign": "center",
-					"boxShadow": "4px 8px 8px rgba(17, 48, 255, 0.1)",
-					"border": "none",
-					"width": "228px",
-					"height": "48px",
-					"display": "flex",
-					"alignItems": "center",
-					"justifyContent": "flex-start",
-					"&:hover": {
-						backgroundColor: "#1130FF",
-					},
-					":hover .headerText": {
-						color: "#FFFFFF",
-					},
-					"borderRadius": "4px",
-					"overflow": "hidden",
-				}}
-				onClick={() => setModalOpen(true)}
-			>
-				<img
-					style={{ marginLeft: "16px" }}
-					src="images/cennznet_blue.svg"
-					alt="CENNZnet-log"
-				/>
-				<Heading
-					className={"headerText"}
-					sx={{
-						fontSize: "16px",
-						color: modalOpen ? "#FFFFFF" : "#1130FF",
-						whiteSpace: "nowrap",
-						textAlign: "center",
-						letterSpacing: "1.2px",
-						ml: "10px",
-						textOverflow: "ellipsis",
-						overflow: "hidden",
-					}}
-				>
-					{selectedAccount?.meta.name || "CONNECT WALLET"}
-				</Heading>
-			</Frame>
+			<div css={styles.walletButton(modalOpen)} onClick={onWalletClick}>
+				<div css={styles.walletIcon}>
+					<img
+						src="images/cennznet_blue.svg"
+						alt="CENNZnet-log"
+						css={styles.walletIconImg}
+					/>
+
+					{!!selectedAccount?.address && (
+						<AccountIdenticon
+							css={styles.walletIconIdenticon}
+							theme="beachball"
+							size={28}
+							value={selectedAccount.address}
+						/>
+					)}
+				</div>
+				<div css={styles.walletState}>
+					{walletState === "Connected" && (
+						<span>{selectedAccount?.meta.name}</span>
+					)}
+					{walletState === "Connecting" && (
+						<span>
+							Connecting
+							<ThreeDots />
+						</span>
+					)}
+					{walletState === "NotConnected" && <span>Connect CENNZNET</span>}
```

Comment:
I think our marketing team would prefer this read `CONNECT CENNZnet`

Reply:
Yip, agree, looks much better! Thanks